client/src/main/java/com/ibm/crail/tools/CrailBenchmark.java

```diff
@@ -739,6 +740,110 @@ void getFileAsync(String filename, int loop, int batch) throws Exception, Interr
 		fs.getStatistics().print("close");
 	}
 	
+	void getMultiFile(String filename, int loop) throws Exception, InterruptedException {
+		final int nrfiles = 16;
+		final int filenamesz = 32;
+
+		System.out.println("getMultiFile, path " + filename);
+
+		//create some files (every instance will create files, but that is ok)
+		String[] chars = {"a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n",
+				"o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z",
+				"0", "1", "2", "3", "4", "5", "6", "7", "8", "9"
+		};
+
+		// Remember my own generated file names
+		String[] fileNames = new String[nrfiles];
+		for (int i = 0; i < nrfiles; i++) {
+			StringBuffer sb = new StringBuffer();
+			sb.append(filename);
+			for (int j = 0; j < filenamesz; j++) {
+				sb.append(chars[ThreadLocalRandom.current().nextInt(0, chars.length)]);
+			}
+			sb.append(".bnc");
+			String file = sb.toString();
+			fs.create(file, CrailNodeType.DATAFILE, CrailStorageClass.DEFAULT, CrailLocationClass.DEFAULT).get().syncDir();
+			String fileNameOnly = file.substring(1);
+			fileNames[i] = file;
+		}
+
+		//benchmark
+		System.out.println("starting benchmark...");
+		fs.getStatistics().reset();
+		double ops = 0;
+		long start = System.currentTimeMillis();
+		while (ops < loop) {
+			ops = ops + 1.0;
+			fs.lookup(fileNames[(int)ops % fileNames.length]).get().asFile();
+		}
+		long end = System.currentTimeMillis();
+		double executionTime = ((double) (end - start)) / 1000.0;
+		double latency = 0.0;
+		if (executionTime > 0) {
+			latency = 1000000.0 * executionTime / ops;
+		}
```

Comment:
I know this logic has been used elsewhere in this file but can we please change this to not use doubles and millis to calculate execution time. Instead use System.nanoTime() and do calculations with long where it makes sense for latency etc we of course need double

Reply:
I am not exactly sure, why calculating in nanoseconds is better than milliseconds (both return long, but we need double for the division).

If we change that, it should be everywhere to keep it consistent. I'll create an issue, as I think that this is a bigger cleanup step.

Reply:
The problem with currentTimeMillis is that it returns the actual time, i.e. is not monotonic and that leads to obvious problems, e.g. if you are running a NTP service time can change while you run a benchmark. That is why it is recommended to use System.nanoTime().

Reply:
Ok sounds good. Will change all benchmark calculations in a new PR.